src/languageFacts/dataProvider.ts

```diff
@@ -109,3 +107,30 @@ export class HTMLDataProvider implements IHTMLDataProvider {
 		return values;
 	}
 }
+
+/**
+ * Generate MarkupContent used in hover/complete
+ * From `documentation` and `references`
+ */
+export function generateMarkupcontent(item: ITagData | IAttributeData | IValueData): MarkupContent {
```

Comment:
IMO it would make sense to have the 'supportMarkdown' flag passed in here so that the content can be streamlined  also for clients that can't handle markdown. E.g. You would then just add the links as is.

Then you don't need an extra step to convert the hover afterwards

Reply:
Maybe find a better name `generateDocumentation(..)` ?